### DISCUSSION TITLE
v3.3.0-beta.6 breaks my addon's test suite - something I'm doing wrong?

### DISCUSSION THREAD

**Discussion by duncanmcclean** [statamic/cms#5497] [Troubleshooting] [answered]
I've been running into the same issue for the last few hours and can't figure out how to fix it... hopefully someone here can help:

<img width="909" alt="image" src="https://user-images.githubusercontent.com/19637309/158266208-b78dc5dd-3320-4b6d-93c8-1b68b068a4f6.png">

I run into this issue when running my addon's test suite on v3.3.0-beta.6 but not when on beta.5 (it passes fine). Seems related to it not being able to find blueprints for the entries.

In my addon test suite, it should be using the blueprints in [this folder](https://github.com/doublethreedigital/simple-commerce/tree/2.4/resources/blueprints/collections), and pulling them in like so in the `TestCase`:

```php
Blueprint::setDirectory(__DIR__ . '/../resources/blueprints');
```

I've tried renaming the blueprint files from `product.yaml` (singular collection name) to `default.yaml` but nada. Tried copying over my blueprints into Orchestra Testbench's `resources/blueprints` folder in case it's reading from there instead but nope.

One commit, everything was passing, the next, most tests were failing due to this issue. Here's an example of a run where I'm seeing the issues: https://github.com/doublethreedigital/simple-commerce/runs/5544587419?check_suite_focus=true

## How to reproduce

1. Clone my addon (Simple Commerce) locally: `git@github.com:doublethreedigital/simple-commerce.git`
2. Checkout the `2.4` branch, `git checkout 2.4`
3. Specifically install the latest beta `composer require statamic/cms:v3.3.0-beta.6`
4. Run the test suite `composer test`
5. You'll see a whole load of failing tests

---

FYI: I created this as a discussion as I'm not sure it's really a bug, more than likely something is setup wrong on my side or needs adjusted to work with 3.3.

**Comment by jasonvarga** [comment, marked as answer]
I noticed that one of your tests creates a `product` entry with a `blueprint: orders`, which doesn't seem right.

Stuff changed on the master branch recently that happens when the files are initially read - you may have just been getting away with the wonky entry silently until now.

**Reply by duncanmcclean** [reply]
Thanks for looking into this! I figured out how to fix it... I moved `Blueprint::setDirectory` into a `Statamic::booted` closure and it started to pick up my blueprints like expected...

See commit: https://github.com/doublethreedigital/simple-commerce/commit/5cada88801567411ce42c69dee93e3a9bb937503
